### ISSUE TITLE
Associate a domain with a column/field of a feature class

### ISSUE THREAD

**Issue by rhewy** [qgis/QGIS#50998]
### Feature description

Add the ability via the GUI/UI to associate a domain with a column/field of a geodatabase

### Additional context

Currently I can make:
  - a  geodatabase
  - a feature-class inside the geodatabase
  - a domain inside the geodatabase
 However, I cannot associate the domain with a column of the feature class

![image](https://user-images.githubusercontent.com/4993063/203846008-2c3b783d-4ece-498a-b3c4-91c45bf074ef.png)

**Comment by nyalldawson** [comment]
This is already possible -- see https://north-road.com/2022/11/24/qgis-3-28-improvements-for-working-with-esri-formats-and-services/

**Comment by rhewy** [comment]
What would cause the fields not to show up for a feature class in the geodatabase

![image](https://user-images.githubusercontent.com/4993063/204098174-34af23db-d350-4e39-ad6e-c627194765e0.png)


![image](https://user-images.githubusercontent.com/4993063/204098214-68433912-c1a1-4d04-8ae4-eabaa02a222c.png)

![image](https://user-images.githubusercontent.com/4993063/204098266-a5a72089-ac28-426f-a92b-e7312a55a3ba.png)

**Comment by rhewy** [comment]
Found it. You cannot have a field called name.

**Comment by rhewy** [comment]
You might want to avoid these columns names too:

https://support.esri.com/en/technical-article/000010906

**Comment by nyalldawson** [comment]
@rhewy 

> Found it. You cannot have a field called name.

Where did you determine that from? ArcGIS Pro seems quite happy to create fields of "name" / "NAME". (Vs the others from that list)

**Comment by rhewy** [comment]
Might not be "name" after all. This is what happens on 3.28.1 with gdal 3.6

Make an ESRI geo database called test.gdb
Make a new table poi::point with fields"
  - name::text
  - poi_type::int32

"Fields"  under the poi layers shows up but not for long.

Then add a domain called poi_types::int32
Add 3 coded values

"Fields"  under the poi layer disappears.
This is why I could not associate the domain with the field as the tree-expand-collapse pointer/arrow goes missing and I cannot access fields to then add the domain.

![image](https://user-images.githubusercontent.com/4993063/204214645-cccfa684-ac6e-4be8-b3fd-cb455f349f8c.png)

I will try making the domain first. Maybe the order that I created things causes the problem.

More fun, I refreshed the test.gdb and the 'Fields' disappeared on both:
![image](https://user-images.githubusercontent.com/4993063/204215864-d258faa7-b8ce-4db6-a579-1a670bebe880.png)

Maybe its a tree-widget issue.
